src/omp/backend.f90

```diff
@@ -291,6 +293,40 @@ subroutine reorder_omp(self, u_, u, direction)
       class(field_t), intent(inout) :: u_
       class(field_t), intent(in) :: u
       integer, intent(in) :: direction
+      integer :: ndir_loc, ndir_groups
+      integer :: i, j, k
+      integer :: out_i, out_j, out_k
+
+      select case (direction)
+         case (RDR_X2Y)
+            ndir_loc = self%xdirps%n
+            ndir_groups = self%xdirps%n_blocks
+         case (RDR_X2Z)
+            ndir_loc = self%xdirps%n
+            ndir_groups = self%xdirps%n_blocks
+         case (RDR_Y2X)
+            ndir_loc = self%ydirps%n
+            ndir_groups = self%ydirps%n_blocks
+         case (RDR_Y2Z)
+            ndir_loc = self%ydirps%n
+            ndir_groups = self%ydirps%n_blocks
+         case (RDR_Z2X)
+            ndir_loc = self%zdirps%n
+            ndir_groups = self%zdirps%n_blocks
+         case (RDR_Z2Y)
+            ndir_loc = self%zdirps%n
+            ndir_groups = self%zdirps%n_blocks
+      end select
+
+      do k=1, ndir_groups
+         do j=1, ndir_loc
+            do i=1, SZ
+               call get_index_reordering(out_i, out_j, out_k, i, j, k, direction, &
+                                         SZ, self%xdirps%n, self%ydirps%n, self%zdirps%n)
+               u_%data(out_i, out_j, out_k) = u%data(i,j,k)
```

Comment:
We needed the mappings between Cartesian and application storage so its great we have them. And using them is the quickest way to have reorders ready. My only concern is the performance of the data copies here. A `$omp parallel do` at the top loop will definitely make things better (maybe with a `collapse(2)` statement as well), and probably is the most we can do without writing specific functions. I think we'll need to check their performance at some point and decide what to do.

Reply:
Agreed, for the `parallel do` I forgot it. Though not sure what gain `collapse(2)` could bring as the loop is perfectly balanced in terms of work in each iteration and the thread count should be small compared to the number of iterations. But it doesn't hurt in putting it there.

Reply:
added the `parallel do`, and I also created an issue to track this #78